src/wallet/wallet.h

```diff
@@ -1405,6 +1411,20 @@ class CWallet final : public FillableSigningProvider, private interfaces::Chain:
 
     /** Implement lookup of key origin information through wallet key metadata. */
     bool GetKeyOrigin(const CKeyID& keyid, KeyOriginInfo& info) const override;
+
+    /** Get last block processed height */
+    int GetLastBlockHeight() const EXCLUSIVE_LOCKS_REQUIRED(cs_wallet)
+    {
+        AssertLockHeld(cs_wallet);
+        return m_last_block_processed_height;
```

Comment:
In commit "Add m_last_block_processed_height field in CWallet" (9161f0bf0321e90e0aaeeafc58c5ea6f97ebb1e2):

Would `assert(m_last_block_processed_height >= 0)` to make sure we're past initialization and don't wind up with crazy depth values.

Reply:
Done